### ISSUE TITLE
Rule: Spaces inside of brackets

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#631]
We need a rule that enforces or disallows spaces inside of parens. It should accept one option, "always" to enforce spaces inside of parens:

``` js
obj[ foo ]
```

And "never", which would enforce no spaces:

``` js
obj[foo]
```

The rule ID should be `space-in-brackets` and should be on with "never" as the default option.
